python/mpt/src/mpt/hash_diff.cairo

```diff
@@ -47,3 +59,97 @@ func poseidon_storage_diff{poseidon_ptr: PoseidonBuiltin*}(diff: StorageDiffEntr
     let (storage_diff_hash) = poseidon_hash_many(5, buffer);
     return storage_diff_hash;
 }
+
+func hash_account_diff_segment{poseidon_ptr: PoseidonBuiltin*}(account_diff: AccountDiff) -> felt {
+    alloc_locals;
+    let len = account_diff.value.len;
+    if (len == 0) {
+        // TODO: Do we return 0 as the hash of an empty segment?
+        // We don't want to raise here as a block could have 0 account diffs.
+        return 0;
+    }
+    let accumulator = poseidon_account_diff(account_diff.value.data[0]);
+    let final_hash = _hash_account_diff_inner(accumulator, account_diff.value.data, 1, len);
+    return final_hash;
+}
+
+func _hash_account_diff_inner{poseidon_ptr: PoseidonBuiltin*}(
+    accumulator: felt, start_ptr: AddressAccountDiffEntry*, i: felt, len: felt
+) -> felt {
+    if (i == len) {
+        return accumulator;
+    }
+    let next_hash = poseidon_account_diff(start_ptr[i]);
+
+    let (buffer) = alloc();
+    assert buffer[0] = accumulator;
+    assert buffer[1] = next_hash;
```

Comment:
at `i=0` here you hash diff[0] with itself right?

Reply:
no since i start at i = 1